Prose/ProseLib/Sources/SidebarFeature/SidebarModel.swift

```diff
@@ -6,16 +6,60 @@
 //  Copyright © 2022 Prose. All rights reserved.
 //
 
+import ConversationFeature
 import SharedModels
+import UnreadFeature
 
-public enum Route: Hashable {
-    case unread, replies, directMessages, peopleAndGroups
-    case chat(id: ChatID)
+public enum SidebarRoute: Hashable {
+    case unread(UnreadScreenModel), replies, directMessages, peopleAndGroups
+    case chat(ConversationState)
     case newMessage
+
+    public func hash(into hasher: inout Hasher) {
+        switch self {
+        case .unread:
+            hasher.combine(0)
+        case .replies:
+            hasher.combine(1)
+        case .directMessages:
+            hasher.combine(2)
+        case .peopleAndGroups:
+            hasher.combine(3)
+        case .chat:
+            hasher.combine(10)
+        case .newMessage:
+            hasher.combine(20)
+        }
+    }
+
+    public static func caseEqual(lhs: Self, rhs: Self) -> Bool {
```

Comment:
Where is that used?

Reply:
By the way, about your question on the difference between the two types of routes in the sidebar, the main issue is this one (I’m having the same problem as yesterday):
In `SidebarContentState`, there is
```swift
public init(
        route: SidebarRoute? = .unread(.init()),
```
In `SpotlightSectionState`, we define the tags of each row
```swift
public struct SpotlightSectionState: Equatable {
    let items: [SidebarItem] = [
        .init(id: .unread(.init()), title: l10n.unreadStack, image: Icon.unread.rawValue, count: 0),
        .init(id: .replies, title: l10n.replies, image: Icon.reply.rawValue, count: 5),
        .init(id: .directMessages, title: l10n.directMessages, image: Icon.directMessage.rawValue, count: 0),
        .init(id: .peopleAndGroups, title: l10n.peopleAndGroups, image: Icon.group.rawValue, count: 2),
    ]
```
When the app starts, it works, since `.unread(.init()) == .unread(.init())`.
However, when we start mutating the state, route changes, but not the SidebarItem.id.
Selecting the row doesn’t work anymore.
That’s why I created the custom `hash(into:)` implementation (which is wrong because it should match the `==` implementation)